jre_emul/Classes/J2ObjC_common.m

```diff
@@ -474,4 +474,18 @@ void unsupportedAdapterCallWithName(NSString *name) {
             sourceSize:sourceArray.count];
 }
 
+IOSObjectArray* toIOSObjectArray(id sourceArray) {
+    J2ObjCKotlinArray<id> *kotlinSourceArray = (J2ObjCKotlinArray<id> *)sourceArray;
```

Comment:
So you're assuming it's a `J2ObjCKotlinArray<id>` an forcing the cast it instead of having `J2ObjCKotlinArray<id>` as the parameter?

Reply:
The issue is that it's not a J2ObjCKotlinArray that is used when called but the implementation that the project version of the kotlin code provided, in the Tests it's CommonKotlinArray and in bell it will be FonseCoreKotlinArray. So i just cast it to the an internal interface that matches it.